quickwit-cli/src/split.rs

```diff
@@ -115,12 +187,113 @@ impl SplitCliCommand {
 
     pub async fn execute(self) -> anyhow::Result<()> {
         match self {
+            Self::List(args) => list_split_cli(args).await,
             Self::Describe(args) => describe_split_cli(args).await,
             Self::Extract(args) => extract_split_cli(args).await,
         }
     }
 }
 
+pub async fn list_split_cli(args: ListSplitArgs) -> anyhow::Result<()> {
+    debug!(args = ?args, "list-split");
+
+    let metastore_uri_resolver = MetastoreUriResolver::default();
+    let metastore = metastore_uri_resolver.resolve(&args.metastore_uri).await?;
+    let time_range_opt = match (args.from, args.to) {
+        (None, None) => None,
+        (None, Some(to)) => Some(Range {
+            start: i64::MIN,
+            end: to,
+        }),
+        (Some(from), None) => Some(Range {
+            start: from,
+            end: i64::MAX,
+        }),
+        (Some(from), Some(to)) => Some(Range {
+            start: from,
+            end: to,
+        }),
+    };
+    let is_disjoint_time_range = |left: &Range<i64>, right: &RangeInclusive<i64>| {
+        left.end <= *right.start() || *right.end() < left.start
+    };
+    let filter_tag_set = args.tags.iter().cloned().collect::<BTreeSet<_>>();
+
+    let mut splits = vec![];
+    // apply tags & time range filter.
+    for split in metastore.list_all_splits(&args.index_id).await? {
+        let is_any_tag_not_in_split = filter_tag_set.iter().any(|tag| {
+            let has_many_tags_for_field = tag
+                .split_once(":")
+                .map(|(field_name, _)| {
+                    split
+                        .split_metadata
+                        .tags
+                        .contains(&format!("{}:*", field_name))
+                })
+                .unwrap_or(false);
+            !(split.split_metadata.tags.contains(tag) || has_many_tags_for_field)
+        });
+        if is_any_tag_not_in_split {
+            continue;
+        }
+
+        if let (Some(filter_time_range), Some(split_time_range)) =
+            (&time_range_opt, &split.split_metadata.time_range)
+        {
+            if is_disjoint_time_range(filter_time_range, split_time_range) {
+                continue;
+            }
+        }
+        splits.push(split);
+    }
+
+    // apply SplitState filter.
+    if !args.states.is_empty() {
+        splits = splits
+            .into_iter()
+            .filter(|split| args.states.contains(&split.split_state))
+            .collect::<Vec<_>>();
+    }
+
+    let mut stdout_handle = stdout();
+    let mut printer = Printer {
+        stdout: &mut stdout_handle,
+    };
+    for split in splits {
```

Comment:
Can we use a library to display the splits as a table? There's an example for `quickwit source list`.